### PR TITLE
Allow re-running of workflows 

### PR DESCRIPTION
**Pull request by ErikSchierboom**
- Don't open a new issue if it already exists
- Prevent pull request from being created more than once

### CODE REVIEW

File: .github/workflows/sync-rest.yml [hunk 1]
```diff
@@ -177,10 +216,12 @@ jobs:
           echo "::set-output name=name::$branch"
 
       - name: Copy globally synced files to target repo
+        if: steps.pr-already-exists.outputs.result == 'false'
         run: |
           cp -a main/global-files/. track-repo/
 
       - name: Determine if repo is a tooling repo
+        if: steps.pr-already-exists.outputs.result == 'false'
```
Comment (ErikSchierboom) [inline .github/workflows/sync-rest.yml:224]: I've not added this check to the steps that check for the repo being a tooling or track repo, as `steps.is-tooling-repo.outputs.result == 'true'` will be false anyway due to the step where that result is used. The same goes for the last steps that all check for `steps.changes.outputs.changes == 'true'`, which will always be false due to the step to set that to true not being executed if the PR already exists

**Review by ErikSchierboom — COMMENTED** [review]
Note that I haven't yet tested the workflow, but I wanted to get feedback first

File: .github/workflows/sync-rest.yml [hunk 2]
```diff
@@ -38,10 +38,20 @@ jobs:
         with:
           result-encoding: string
           script: |
+            const title = `[Tracking] ${context.workflow} - ${context.eventName} - ${context.actor} - ${context.sha}`
+            const existingIssue = (await octokit.paginate(octokit.rest.issues.listForRepo, {
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+            })).find((issue) => issue.title === title)
```
Comment (SaschaMann) [inline .github/workflows/sync-rest.yml:45]:
```suggestion
            const existingIssue = (await octokit.paginate(octokit.rest.issues.listForRepo, {
              owner: context.repo.owner,
              repo: context.repo.repo,
              labels: '👁 tracking issue',
              state: 'open',
            })).find((issue) => issue.title === title)
```

File: .github/workflows/sync-rest.yml [hunk 3]
```diff
@@ -151,12 +161,40 @@ jobs:
...
+        with:
+          github-token: ${{ secrets.BOT_PERSONAL_ACCESS_TOKEN }}
+          script: |
+            const repo = process.env.TARGET_REPO_FULLNAME.split('/')[1]
+
+            return (
+              await octokit.paginate(octokit.rest.issues.listForRepo, {
+                owner: "exercism",
+                repo: repo,
+              })
```
Comment (SaschaMann) [inline .github/workflows/sync-rest.yml:179]:
```suggestion
              await octokit.paginate(octokit.rest.issues.listForRepo, {
                owner: "exercism",
                repo: repo,
                state: 'open',
              })
```

**Review by SaschaMann — COMMENTED** [review]
Would be best to test this somewhere but it looks good to me. Limiting the `listForRepo` results with the right label and state should save us some API calls because we're probably not going to run into pagination.

**Comment by ErikSchierboom** [comment]
@SaschaMann 
This is the issue creation being skipped: https://github.com/exercism/org-wide-files/runs/5464765618?check_suite_focus=true 
This is the PR being created: https://github.com/exercism/org-wide-files/runs/5463961328?check_suite_focus=true
This is the PR being skipped: https://github.com/exercism/org-wide-files/runs/5464601595?check_suite_focus=true
This is a new run with a new repo as its argument: https://github.com/exercism/org-wide-files/runs/5464797097?check_suite_focus=true

File: .github/workflows/sync-rest.yml [hunk 4]
```diff
@@ -38,12 +38,26 @@ jobs:
...
+
+            const existingIssue = (await octokit.paginate(octokit.rest.issues.listForRepo, {
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              labels: label,
+              state: 'open',
+            })).find((issue) => issue.title === title)
+
+            if (existingIssue) {
+              return existingIssue.html_url
```
Comment (SaschaMann) [inline .github/workflows/sync-rest.yml:52]:
```suggestion
              console.log(`::notice::Existing issue found: ${existingIssue.html_url}`)
              return existingIssue.html_url
```

File: .github/workflows/sync-rest.yml [hunk 5]
```diff
@@ -151,12 +165,41 @@ jobs:
...
+
+            return (
+              await octokit.paginate(octokit.rest.issues.listForRepo, {
+                owner: "exercism",
+                repo: repo,
+                state: 'open',
+              })
+            ).some((pr) =>
+              pr.body.includes(process.env.TRACKING_ISSUE_URL)
+            )
```
Comment (SaschaMann) [inline .github/workflows/sync-rest.yml:187]: I'm not familiar with this syntax, could you add a log message here too in case a PR was found?

**Review by SaschaMann — COMMENTED** [review]
Those checks look good.

**Comment by SaschaMann** [comment]
Needs a rebase

**Comment by ErikSchierboom** [comment]
Rebased

**Pull request merged**